### ISSUE TITLE
Add mechanism for setting common call params across providers

### ISSUE THREAD

**Issue by barapa** [Mirascope/mirascope#667]
### Description

When calling the same prompt across multiple providers, if you want to set model config values such as temperature, max_output_tokens, etc, each provider's types are different. Additionally, different providers have different values that need to be passed with no defaults.

For example:
```py
from google.generativeai.types.generation_types import GenerationConfig
from mirascope.core import anthropic, gemini, openai, prompt_template


@prompt_template(
    """
    SYSTEM:
    Do what you are told.

    USER:
    {what_to_do}
    """
)
def prompt(
    what_to_do: str,
): ...


def main():
    prompt_fn = prompt

    openai_call = openai.call("gpt-4o", call_params={"temperature": 0.7})(prompt_fn)
    openai_result = openai_call(what_to_do="Do something cool")

    # Anthropic requires max_tokens to be set
    anthropic_call = anthropic.call(
        "claude-3-opus-latest",
        call_params={
            "temperature": 0.7,
            "max_tokens": 2048,
        },
    )(prompt_fn)
    anthropic_result = anthropic_call(what_to_do="Do something cool")

    # Gemini requires its own specific type
    gemini_call = gemini.call(
        model="gemini-1.5-pro-002",
        call_params={
            "generation_config": GenerationConfig(temperature=0.7, max_output_tokens=2048)
        },
    )(prompt_fn)
    gemini_result = gemini_call(what_to_do="Do something cool")


if __name__ == "__main__":
    main()


```

It might be useful to have some common way to generate the equivalent sets of params for each provider.

**Comment by willbakst** [comment]
Totally! We originally implemented things this way because each provider has different call parameters, but there are certain call parameters that are common, so convenience around that would certainly be nice.

I'm thinking something like this would be best:

```python
openai_call_params = openai.OpenAICallParams.from_common(temperature=0.7)
openai_call = openai.call("gpt-4o", call_params=openai_call_params)(prompt_fn)
openai_result = openai_call(what_to_do="Do something cool")

anthropic_call_params = anthropic.AnthropicCallParams.from_common(temperature=0.7)
anthropic_call = anthropic.call("claude-3-opus-latest", call_params=anthropic_call_params)(prompt_fn)
anthropic_result = anthropic_call(what_to_do="Do something cool")

gemini_call_params = gemini.GeminiCallParams.from_common(temperature=0.7)
gemini_call = gemini.call(model="gemini-1.5-pro-002", call_params=gemini_call_params)(prompt_fn)
gemini_result = gemini_call(what_to_do="Do something cool")
```

What do you think?

**Comment by willbakst** [comment]
We opted for simply allowing users to pass in a `CommonCallParams` dictionary and it will work :)

This will be released on Friday

**Comment by willbakst** [comment]
This is released in `v1.9`! (with a quick follow up fix in `v1.9.1` for a type error I found, oops)

**Comment by willbakst** [comment]
and....another quick fix in `v1.9.2` to fix an issue with max tokens missing for anthropic 🤦
